src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs

```diff
@@ -179,12 +173,8 @@ private static AzureCredentials GetServicePrincipleCredentials(AzureEnvironment
 
         private static AzureCredentials GetUserAssignedManagedIdentityCredentials(AzureEnvironment azureCloud, string tenantId, AzureAuthenticationInfo azureAuthenticationInfo, AzureCredentialsFactory azureCredentialsFactory)
         {
-            if (string.IsNullOrWhiteSpace(azureAuthenticationInfo.IdentityId))
-            {
-                throw new AuthenticationException("No identity was configured for user-assigned managed identity");
-            }
-
-            return azureCredentialsFactory.FromUserAssigedManagedServiceIdentity(azureAuthenticationInfo.IdentityId, MSIResourceType.VirtualMachine, azureCloud, tenantId);
+            var clientId = string.IsNullOrWhiteSpace(azureAuthenticationInfo.IdentityId) ? null : azureAuthenticationInfo.IdentityId;
```

Comment:
Since we have this logic multiple times, we might want to do this:
```suggestion
            var clientId = azureAuthenticationInfo.DetermineIdentityId();
```

Reply:
Created `GetIdentityIdOrDefault` that accepts a default value to support this case of null and other cases like "externally-configured-user-assigned-identity"